service/src/main/java/bio/terra/catalog/service/DatasetService.java

```diff
@@ -181,6 +187,39 @@ private DatasetPreviewTable generateDatarepoTable(
                 .getResult());
   }
 
+  private DatasetPreviewTable generateRawlsTable(
+      AuthenticatedUserRequest user, Dataset dataset, String tableName) {
+    Map<String, EntityTypeMetadata> ent =
+        rawlsService.entityMetadata(user, dataset.storageSourceId());
+    EntityTypeMetadata obj = ent.get(tableName);
+    return new DatasetPreviewTable()
+        .columns(convertEntityToColumn(obj))
+        .rows(
+            rawlsService
+                .entityQuery(user, dataset.storageSourceId(), tableName)
+                .getResults()
+                .stream()
+                .map(Entity -> convertEntityToRow(Entity, obj.getIdName()))
+                .toList());
+  }
+
+  private Object convertEntityToRow(Entity entity, String idName) {
+    @SuppressWarnings("unchecked")
+    Map<String, String> att = (Map<String, String>) entity.getAttributes();
+    Map<String, String> rows = new HashMap<>(att);
+    rows.put(idName, entity.getName());
+    return rows;
+  }
+
+  private static List<ColumnModel> convertEntityToColumn(EntityTypeMetadata entity) {
```

Comment:
This is actually converting the entity type metadata (which is the table metadata) to columns. So maybe the name should be `convertTableMetadataToColumns`?